### PR TITLE
[event_v2] migrate the rest files in aptos-framework

### PR DESCRIPTION
**Pull request by lightmark**
### Description
migrate other files

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/block.move [hunk 1]
```diff
@@ -163,7 +190,17 @@ module aptos_framework::block {
...
+            hash,
+            epoch,
+            round,
+            height: block_metadata_ref.height,
+            previous_block_votes_bitvec,
+            proposer,
+            failed_proposer_indices,
+            time_microseconds: timestamp,
+        };
+        emit_new_block_event(vm, &mut block_metadata_ref.new_block_events, new_block_event, new_block_event_v2);
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/block.move:203]: if we're doing this, we might incur a performance impact on each block before we even migrate.

File: aptos-move/framework/aptos-framework/sources/staking_contract.spec.move [hunk 2]
```diff
@@ -143,30 +142,30 @@ spec aptos_framework::staking_contract {
...
-        voter: address,
-        amount: u64,
-        commission_percentage: u64,
-        contract_creation_seed: vector<u8>,
+    staker: &signer,
+    operator: address,
+    voter: address,
+    amount: u64,
+    commission_percentage: u64,
+    contract_creation_seed: vector<u8>,
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/staking_contract.spec.move:150]: something weird

**Review by davidiw — APPROVED** [review]
would be good to test this with an updated indexer processor and verify it is correct before we end up shipping this.

also might be worth adding the disabling of v1 in there as well for that reason.

File: aptos-move/framework/aptos-framework/sources/staking_contract.spec.move (hunk 2 above)
Comment (lightmark) [reply aptos-move/framework/aptos-framework/sources/staking_contract.spec.move:150]: [intellij](https://www.google.com/search?sca_esv=abe270d53af57e7a&rlz=1C5GCCM_en&sxsrf=ACQVn0-2uIhyp62gtcxL8CPsV8EeuG9v9w:1711694967004&q=intellij&spell=1&sa=X&ved=2ahUKEwiy8OLT8JiFAxXJpVYBHUN8AuYQkeECKAB6BAgGEAI) move plugin does this automatically and I have no way to disable it...

File: aptos-move/framework/aptos-framework/sources/block.move (hunk 1 above)
Comment (lightmark) [reply aptos-move/framework/aptos-framework/sources/block.move:203]: one more event per block? should not be an issue regarding perf.

**Comment by lightmark** [comment]
I will rebase once #11224 lands

**Comment by lightmark** [comment]
@grao1991 do you need to change the new block event tracking after this PR?

**Comment by grao1991** [comment]
> @grao1991 do you need to change the new block event tracking after this PR?

Yes. It's protected by the sharding flag which hasn't been enabled. You go first and I can try to make the change later.

FYI, you need to double check if other things (e.g. consensus) that depending on the new block event works correctly after your change.

**Review by bowenyang007 — APPROVED** [review]

**Comment by bowenyang007** [comment]
When are we removing the old events though?

**Comment by lightmark** [comment]
> When are we removing the old events though?

Let's do it after 1-2 release.

**Pull request merged**